samvera-labs/clover-iiif
# Update dependencies

## What does this do?

There were some major dependency version lags, so this PR attempted to upgrade all deps.  However, after upgrading to Vite `v5`, the TypeScript types generation as part of the build process broke.  We're using `dts-bundle-generator` to handle this.  I tried updating the build to use `vite-plugin-dts`, a more popular solution in numbers, but that was erroring-out under the hood as well when trying to roll up all types into one file. 

So, the end solution here is to pin to an older version of `dts-bundle-generator`.  

Another byproduct of upgrading to the latest Vite, is that the web-component build is broken as well.  I'll look at this and see if there's a way to compile it, and make sure it doesn't break for anyone using it.   Is anyone using it I wonder tho?

## How to test
Pull in the latest release candidate version into a demo environment:
```bash
pnpm add @samvera/clover-iiif@next
```
And verify
- The Clover components work as expected
- Typescript intellisense is working in your IDE
- You can use the web-component version as well
